src/get-started/create-service.md

```diff
@@ -371,115 +364,147 @@ executed:
+#[derive(Debug, ExecutionFail)]
 pub enum Error {
-    #[fail(display = "Wallet already exists")]
+    /// Wallet already exists.
     WalletAlreadyExists = 0,
-
-    #[fail(display = "Sender does not exist")]
+    /// Sender doesn't exist.
     SenderNotFound = 1,
-
-    #[fail(display = "Receiver does not exist")]
+    /// Receiver doesn't exist.
     ReceiverNotFound = 2,
-
-    #[fail(display = "Insufficient currency amount")]
+    /// Insufficient currency amount.
     InsufficientCurrencyAmount = 3,
-
-    #[fail(display = "Sender same as receiver")]
+    /// Sender same as receiver.
     SenderSameAsReceiver = 4,
 }
-
-// Conversion between service-specific errors and the standard error type
-// that can be emitted by transactions.
-impl From<Error> for ExecutionError {
-    fn from(value: Error) -> ExecutionError {
-        let description = format!("{}", value);
-        ExecutionError::with_description(value as u8, description)
-    }
-}
 ```
 
+Deriving the `ExecutionFail` trait here will make our errors generic and
+compatible with other error kinds used within Exonum blockchain, this macro
+is similar to the `failure::Fail`.
+
 ### Transaction Execution
 
-Every transaction in Exonum has business logic of the blockchain attached,
-which is encapsulated in the `Transaction` trait.
-This trait has `execute` method which contains logic applied to the
-storage when a transaction is executed.
+Above we've defined the interface of our service, but currently this interface
+has no implementation. Thus, there is no actual business logic attached to
+them. To fix this situation, we should declare our service, and then implement
+the `CryptocurrencyInterface` trait for it.
+
+Service is a `struct` which implements several certain traits:
+
+```rust
+/// Cryptocurrency service implementation.
+#[derive(Debug, ServiceFactory, ServiceDispatcher)]
+#[service_dispatcher(implements("CryptocurrencyInterface"))]
+#[service_factory(proto_sources = "crate::proto")]
+pub struct CryptocurrencyService;
+
+impl Service for CryptocurrencyService {}
+```
+
+`service_dispatcher` macro collects information about interfaces implemented
+by service, and `service_factory` macro generates a code to create instances
+of our service, similarly to the [Factory method design pattern][factory-method].
+
+[factory-method]: https://en.wikipedia.org/wiki/Factory_method_pattern
+
+The implementation of the `Service` trait can contain the additional elements of
+the service lifecycle, like wiring the API. Currently we can skip them and
+leave the implementation empty.
 
-In our case `execute` method gets the reference to the `TransactionContext`.
-It includes `Fork` of the storage (can be accessed with `.fork()`) and the
-public key which was used to sign the transaction (can be accessed with
-`.author()`). We wrap `Fork` with our `CurrencySchema` to access our data
-layout.
+Now, when we have the structure, we can implement the actual business logic.
+
```

Comment:
Nit: IMO, this newline is redundant; the following sentence looks fine as a part of the previous paragraph.